pkgs/development/interpreters/python/python-packages-base.nix

```diff
@@ -30,13 +33,17 @@ let
 
       overrideWith = newArgs: args // lib.toFunction newArgs args;
       overridePythonAttrs = mirrorArgs (newArgs: makeOverridablePythonPackage f (overrideWith newArgs));
+
+      # Change the result of the function call by applying g to it
+      overrideResult = g: makeOverridablePythonPackage (mirrorArgs (args: g (f args))) origArgs;
     in
-      if builtins.isAttrs result then result
-      else if builtins.isFunction result then {
-        inherit overridePythonAttrs;
-        __functor = self: result;
-      }
-      else result);
+    if builtins.isAttrs result then result
+      // lib.optionalAttrs (result ? overrideAttrs) { overrideAttrs = fdrv: overrideResult (drv: drv.overrideAttrs fdrv); }
```

Comment:
Don't we need () here or are the redundant?

Reply:
I don't quite get it.

`result ? overrideAttrs` would be true only when `result` is an attribute. Maybe

```nix
if builtins.isAttrs result then
  result // lib.optionalAttrs (result?overrideAttrs) {
    overrideAttrs =...;
  }
else if ...
```

If you are talking about the let-in construct above the if-then-else construct, it surely doesn't need parenthesis.

```
nix-repl> let in if 2 + 2 == 5 then "y" else "n"
"n"
```

Reply:
> Don't we need () here or are the redundant?

Or do you mean that `set?attr` has higher precedence over function application and the parenthesis around it is redundant?